src/cli/bin.js

```diff
@@ -86,19 +63,54 @@ if (args[0] === 'daemon' || args[0] === 'init') {
     if (err) {
       throw err
     }
+
     utils.getIPFS(argv, (err, ipfs, cleanup) => {
-      if (err) { throw err }
+      if (err) {
+        throw err
+      }
+
+      // add mfs commands
+      mfs(cli)
+
+      // NOTE: This creates an alias of
+      // `jsipfs files {add, get, cat}` to `jsipfs {add, get, cat}`.
+      // This will stay until https://github.com/ipfs/specs/issues/98 is resolved.
+      const addCmd = require('./commands/files/add')
+      const catCmd = require('./commands/files/cat')
+      const getCmd = require('./commands/files/get')
+      const aliases = [addCmd, catCmd, getCmd]
+      aliases.forEach((alias) => {
+        cli.command(alias)
+      })
 
       cli
+        .commandDir('commands')
         .help()
         .strict()
         .completion()
-        .parse(args, { ipfs: ipfs }, (err, argv, output) => {
-          if (output) { print(output) }
 
-          cleanup(() => {
-            if (err) { throw err }
-          })
+      const parser = new YargsPromise(cli, { ipfs })
+      parser.parse(args)
+        .then(({ data, argv }) => {
+          if (data) {
+            print(data)
+          }
+
+          return cleanup()
+        })
+        .catch((arg) => {
+          debug(arg)
+
+          // the argument can have a different shape depending on where the error came from
+          if (arg.message) {
+            print(arg.message)
+          } else if (arg.error && arg.error.message) {
+            print(arg.error.message)
+          } else {
+            print('Unknown error, please re-run the command with DEBUG=ipfs:cli to see debug output')
```

Comment:
It would be great to have some consistency here - the rest of the cli debug prefixes are `cli:*` and elsewhere in the code we have `jsipfs:*`. One for a separate PR 😉

Reply:
I'd love it if we could standardise on one prefix for all `ipfs` related `DEBUG` values.  Just doing `ipfs:*` instead of `ipfs:*,ipfsd-ctl:*,ipfs-api:*,etc` is my dream.